vcbuild.bat

```diff
@@ -422,12 +432,56 @@ set USE_EMBEDDED_NODE_INSPECT=1
 goto node-tests
 
 :node-tests
-if "%test_args%"=="" goto cpplint
+if "%test_args%"=="" goto test-v8
 if "%config%"=="Debug" set test_args=--mode=debug %test_args%
 if "%config%"=="Release" set test_args=--mode=release %test_args%
 echo running 'cctest %cctest_args%'
 "%config%\cctest" %cctest_args%
 call :run-python tools\test.py %test_args%
+goto test-v8
+
+:test-v8
+if not defined custom_v8_test goto cpplint
+if not defined DEPOT_TOOLS_PATH goto depot-tools-not-found
+if "%config%"=="Debug" set test_args=%target_arch%.debug
+if "%config%"=="Release" set test_args=%target_arch%.release
+set savedpath=%path%
+set path=%DEPOT_TOOLS_PATH%;%path%
+
+echo calling: tools\make-v8.sh
+sh tools\make-v8.sh
+if errorlevel 1 goto exit
+cd %~dp0
+cd deps\v8
+echo running 'python tools\dev\v8gen.py %test_args%'
+call python tools\dev\v8gen.py %test_args% > nul 2> nul
+if errorlevel 1 goto exit
+echo running 'ninja -C out.gn/%test_args% %v8_build_options%'
+call ninja -C out.gn/%test_args% %v8_build_options% > nul 2> nul
+if errorlevel 1 goto exit
+set path=%savedpath%
+echo %cd%
+if not defined test_v8 goto test-v8-intl
+echo running 'python tools\run-tests.py %common_v8_test_options% %v8_test_options% --junitout ./v8-tap.xml'
+call python tools\run-tests.py %common_v8_test_options% %v8_test_options% --junitout ./v8-tap.xml
+
+:test-v8-intl
+if not defined test_v8_intl goto test-v8-benchmarks
+echo running 'python tools\run-tests.py %common_v8_test_options% intl --junitout ./v8-intl-tap.xml'
+call python tools\run-tests.py %common_v8_test_options% intl --junitout ./v8-intl-tap.xml
+
+:test-v8-benchmarks
+if not defined test_v8_benchmarks goto restorepath
+echo running 'python tools\run-tests.py %common_v8_test_options% benchmarks --junitout ./v8-benchmarks-tap.xml'
+call python tools\run-tests.py %common_v8_test_options% benchmarks --junitout ./v8-benchmarks-tap.xml
+goto restorepath
+
+:depot-tools-not-found
```

Comment:
Second definition here.

Follow on question, as it's only called once, is there a need for a separate `goto` for it at all? IDK what others think, but for me every `goto` adds significant mental complexity, so if it's avoidable I'd like to avoid it.